metaflow/datatools/s3.py

```diff
@@ -316,6 +316,12 @@ def __init__(
                 bucket = parsed.netloc
             if not prefix:
                 prefix = parsed.path
+
+                if DATATOOLS_S3ROOT is None:
+                    raise MetaflowS3URLException(
+                        "No config or correct S3 root path provided when trying to use S3 storage"
+                    )
+
```

Comment:
why only check this inside the `if run.. if not prefix..` ?

Reply:
It is only used in the `if run` portion so IMHO, it could be checked there. I would also check it in `get_root_from_config` (used for `IncludeFile`).

Reply:
@exotikh3 Can you address this comment so that we can proceed with the merge?